### PR TITLE
Add Docker configuration + instructions for developers

### LINKED ISSUE #26: Dockerize Azure function app so that devs with M1s can run it
Add a Dockerfile to the Azure function app so that people with M1s (and plain docker enthusiasts) can run the app in a docker container for local development. (Because Azure Functions is not compatible with ARM architecture)

### PR DESCRIPTION
**Pull request by wbobeirne**
Closes #26.

### What This Does

* Adds a `Dockerfile` using the base python3.8 image, that then installs the necessary Azure CLI tools
* Adds a `docker-compose.yml` that manages configuration for running the docker image in development
* Updates README.md with instructions on how to use it

### Steps to Test

1. Make sure you Docker installed
    * If on an M1, please note the README instructions on specific configurations and versions
1. Run `docker-compose up`, confirm the service starts up OK

### CODE REVIEW

**Comment by DevTrav** [comment]
Approve. The solution works in my M1 environ. Awaiting approval from @zimzoom to merge.

**Review by nareddyt — APPROVED** [review]
Works great for me! Thanks for putting this together

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM  --platform=linux/amd64 python:3.8-slim-bullseye
```
Comment (nareddyt) [inline Dockerfile:1]: @zimzoom mentioned we actually updated to python3.9 but forgot to update the README. Can you update this?

Comment (wbobeirne) [reply Dockerfile]: 👍

**Review by nareddyt — APPROVED** [review]

**Pull request merged**